utility/RTC/RTC.h

```diff
@@ -0,0 +1,75 @@
+/**************************************************************************/
+/*!
+  @file     RTC.h
+
+  Original library by JeeLabs http://news.jeelabs.org/code/, released to the
+  public domain
+
+  License: MIT (see LICENSE)
+
+  This is a fork of JeeLab's fantastic real time clock library for Arduino.
+
+  For details on using this library with an RTC module like the DS1307, PCF8523,
+  or DS3231, see the guide at:
+  https://learn.adafruit.com/ds1307-real-time-clock-breakout-board-kit/overview
+
+  Adafruit invests time and resources providing this open source code,
+  please support Adafruit and open-source hardware by purchasing
+  products from Adafruit!
+*/
+/**************************************************************************/
+
+#ifndef _RTC_H_
+#define _RTC_H_
+
+#include <Arduino.h>
+#include "../DateTime/DateTime.h"
+
+/**************************************************************************/
+/*!
+    @brief  RTC superclass for all RTC chips
+*/
+/**************************************************************************/
+class RTC {
+public:
+  /*!
+    @brief Start the RTC
+    @param dt DateTime object containing desired date/time
+    @return True if successful, false otherwise
+  */
+  virtual boolean begin(DateTime &dt) = 0;
+
+  /*!
+    @brief Adjust the RTC to the specified date/time
+    @param dt DateTime object containing desired date/time
+  */
+  virtual void adjust(const DateTime &dt) = 0;
+
+  /*!
+    @brief Adjust the RTC's date/time to account for RTC drift
+    @param drift The drift to adjust the date/time by
+    @note Positive values makes the clock go ahead in time and vice-versa
+  */
+  virtual void adjustDrift(const int drift);
```

Comment:
This looks like an attempt to generalize `RTC_Micros::adjustDrift()`, but you seem to have misunderstood the purpose of that method. Its purpose is not to “Adjust the date/time to account for the drift”, it is instead to adjust the drift itself, or the drift _rate_ if you wish. This adjustment does not make the time “jump” forward or back, it instead makes the clock run faster or slower.

Reply:
Will fix in another pull request for another branch dedicated to implementing `adjustDrift()` for every concrete RTC class